### PR TITLE
Fix failing test due to a bug in NumPy when using OpenBLAS

### PR DESCRIPTION
**Pull request by lezcano**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #67679

implementations

Fixes https://github.com/pytorch/pytorch/issues/67675

cc @mruberry

Differential Revision: [D32368698](https://our.internmc.facebook.com/intern/diff/D32368698)

### CODE REVIEW

File: test/test_linalg.py [hunk 1]
```diff
@@ -1638,7 +1638,12 @@ def run_test_case(input, p):
         a = torch.eye(3, dtype=dtype, device=device)
         a[-1, -1] = 0  # make 'a' singular
         for p in norm_types:
-            run_test_case(a, p)
+            try:
+                run_test_case(a, p)
+            except numpy.linalg.LinAlgError:
+                # Numpy may fail to converge for some BLAS backends (although this is very rare)
+                # See the discussion in #67675
```
Comment (IvanYashchuk) [inline test/test_linalg.py:1645]: It's not clear whether it's referring to a PR or an issue and in which repository NumPy or PyTorch.
```suggestion
                # See the discussion in https://github.com/pytorch/pytorch/issues/67675
```

**Review by IvanYashchuk — COMMENTED** [review]
It's possible to get the information about which LAPACK implementation is used in NumPy with `numpy.show_config()`. Maybe the test should pass only when OpenBLAS is used.

**Comment by Lezcano** [comment]
I've looked into `numpy.show_config()` but it looks like parsing it would make this into a very brittle test. I vote for leave the fix as it is, as I don't think that we'll encounter many more NumPy bugs in that particular piece of code.

File: test/test_linalg.py [hunk 2]
```diff
@@ -1638,7 +1638,12 @@ def run_test_case(input, p):
         a = torch.eye(3, dtype=dtype, device=device)
         a[-1, -1] = 0  # make 'a' singular
         for p in norm_types:
-            run_test_case(a, p)
+            try:
+                run_test_case(a, p)
+            except numpy.linalg.LinAlgError:
```
Comment (casparvl) [inline test/test_linalg.py:1643]: numpy is imported as `np`, so this should be `np.linalg.LinAlgError`

Comment (Lezcano) [reply test/test_linalg.py:1643]: ooops!

**Comment by Lezcano** [comment]
@mruberry this should be ready.

**Comment by mruberry** [comment]
Hey @Lezcano this looks great but looks like it picked up some kineto updates, too.

Would you remove those and just reping me to merge this?

**Comment by Lezcano** [comment]
Submodules are my worst enemy, or at least, certainly in the top 5.
Fixed @mruberry

**Review by mruberry — APPROVED** [review]
Cool! Thanks @Lezcano

**Comment by mruberry** [comment]
@mruberry has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32368698).

**Pull request closed without merging**